yt/funcs.py

```diff
@@ -77,16 +98,11 @@ def ensure_numpy_array(obj):
 
 def ensure_tuple(obj):
     """
-    This function ensures that *obj* is a tuple.  Typically used to convert
+    This function ensures that *obj* is a tuple. Typically used to convert
     scalar, list, or array arguments specified by a user in a context where
     we assume a tuple internally
     """
-    if isinstance(obj, tuple):
-        return obj
-    elif isinstance(obj, (list, np.ndarray)):
-        return tuple(obj)
-    else:
-        return (obj,)
+    return tuple(always_iterable(obj))
```

Comment:
since we have this update of ensure_tuple() that does `return tuple(always_iterable(obj))`, is it worth adding back the `ensure_list()` functionality that acts as a sanitizer and and use that in the code for consistency?

Reply:
actually my plan was to get rid of both. As far as I recall, `ensure_tuple` is only ever used to sanitize `Dataset.perdiodicity` attributes _within the code base_, and what's more is that I _think_ it's only ever done in the Boxlib frontend, so I would rather just get this one down that reintroduce its sibling. Does this sound fair to you ?

Reply:
(sorry about not replying to your initial comment on this btw, it slipped my mind so it seems.)

Reply:
> (sorry about not replying to your initial comment on this btw, it slipped my mind so it seems.) 

Not a problem! I just wanted to make sure it didn't get lost! 

> I would rather just get this one down that reintroduce its sibling. Does this sound fair to you ? 

I think that's totally fine. 🙂 I was just curious for consistency reasons.

Reply:
I'll remove `ensure_tuple` in a followup PR !

Reply:
SGTM!